### PR TITLE
Make fit_xy() fail when mode is unknown

### PR DESCRIPTION
**Pull request by topepo**
As `fit()` currently does.

### CODE REVIEW

**Review by DavisVaughan — APPROVED** [review]
It is probably worth testing this against the book and the workshop notes? It seems like something that people could definitely accidentally rely on.

Also, I think you should be able to remove every usage of `check_mode()`, because mode is now required in all cases and is never inferred "automatically"
https://github.com/tidymodels/parsnip/search?q=check_mode

File: parsnip.Rproj [hunk 1]
```diff
@@ -17,5 +17,6 @@ StripTrailingWhitespace: Yes
 
 BuildType: Package
 PackageUseDevtools: Yes
+PackageCleanBeforeInstall: Yes
```
Comment (DavisVaughan) [inline parsnip.Rproj:20]: Looks like you need to update your version of the IDE. Neither the "released" version of the IDE nor the current daily use this anymore

**Comment by topepo** [comment]
Checked against TMwR and revdeps. Everything looks good.

**Comment by DavisVaughan** [comment]
Ok, I think we are good to move forward with this then. I'll go ahead and release workflows with this change.

**Pull request merged**